app/actors/KafkaDatasetStatusClient.java

```diff
@@ -0,0 +1,132 @@
+package actors;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import configuration.Configuration;
+import exceptions.DatasetStatusException;
+import models.DatasetStatus;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import play.Logger;
+import services.DatasetStatusUpdater;
+import utils.KafkaUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import static java.util.Collections.singletonList;
+
+/**
+ * Kafka client capable of sending and receiving messages related to dataset status.
+ */
+public class KafkaDatasetStatusClient {
+
+    private static final long DEAD_DATASET_THRESHOLD = 1000l * 60 * 15; // 15 minutes of inactivity
```

Comment:
There's also the slightly more hip Java 8 version: `Duration.ofMinutes(15).toMillis()`